code/scripting/api/libs/base.cpp

```diff
@@ -481,6 +483,41 @@ ADE_FUNC(XSTR,
 	return ade_set_args(L, "s", translated.c_str());
 }
 
+ADE_FUNC(translateTokens, 
+	l_Base, 
+	"string text", 
+	"Returns a string that translates any default control binding to current binding (same as Directive Text). Default binding must be encapsulated by '$$' to be translated.",
+	"string",
+	"Translated string or nil if invalid")
+{
+	const char* untranslated_str;
+	if (!ade_get_args(L, "s", &untranslated_str)) {
+		return ADE_RETURN_NIL;
+	}
+
+	SCP_string translated_str = message_translate_tokens(untranslated_str);
+
+	return ade_set_args(L, "s", translated_str.c_str());
+}
+
+ADE_FUNC(replaceVariableValue,
```

Comment:
I would call this `translateVariables`, to be consistent with the other function name.  Or maybe a better idea would be to use `replaceTokens` and `replaceVariables`.

Reply:
Good idea, updated names.